docs/ADR-014-Total-conversion-functions-conventions.md

```diff
@@ -0,0 +1,70 @@
+# Status
+
+- [ ] Proposed 2024-05-21
+
+# Context
+
+## The Problem
+In `cardano-api` we have multiple functions performing conversions between one value of the type to the other, for example:
+
+```haskell
+toShelleyScriptHash :: ScriptHash -> ScriptHash
+fromShelleyScriptHash :: ScriptHash -> ScriptHash
+```
+
+There are multiple naming conventions for the conversion functions which makes them hard to locate.
+Some conversion functions with lengthy names, are not very convenient to use.
+
+## Solution Proposal
+
+### Type classes
+
+For total functions, which are simply converting a value from one type to another, we can use type classes [`Inject`](https://cardano-ledger.cardano.intersectmbo.org/cardano-ledger-core/Cardano-Ledger-BaseTypes.html#t:Inject) & [`Convert`](https://cardano-api.cardano.intersectmbo.org/cardano-api/Cardano-Api-Internal-Eras.html#t:Convert):
```

Comment:
Needs explanation why we need `Convert`. (vide @newhoggy)

Reply:
`Convert` is less ambiguous then `Inject` and it was implemented for use with our eon conversions.